### PR TITLE
LPS-72027 Review of Brian's SF changes

### PR DESCRIPTION
**Pull request by ahdezma**

### CODE REVIEW

**Comment by ahdezma** [comment]
ci:retest

**Comment by ahdezma** [comment]
ci:retest

File: vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/internal/InvalidGenericException.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.liferay.vulcan.wiring.osgi.internal;
+
+/**
+ * @author Alejandro Hernández
+ * @author Carlos Sierra Andrés
+ * @author Jorge Ferrer
+ */
+public class InvalidGenericException extends RuntimeException {
```
Comment (JorgeFerrer) [inline vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/internal/InvalidGenericException.java]: Brian suggested that we follow the new pattern in portal of using a generic exception (such as VulcanException, although I would prefer something a bit more specific) and then nested exceptions for the specific error (such as MustHaveGeneric).

What do you think?

File: vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/RepresentorManager.java [hunk 2]
```diff
@@ -215,14 +215,15 @@ public RepresentorManager() {
 	}
 
 	private final BundleContext _bundleContext;
-	private final Map<String, List<EmbeddedTuple<?, ?>>> _embeddedTuples =
+	private final Map<String, List<EmbeddedTuple<?, ?>>> _embeddedTupleLists =
```
Comment (JorgeFerrer) [inline vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/RepresentorManager.java]: Why this change?

File: vulcan-liferay-api/src/main/java/com/liferay/vulcan/liferay/internal/provider/CompanyContextProvider.java [hunk 3]
```diff
@@ -34,10 +34,7 @@
  * @author Carlos Sierra Andrés
  * @author Jorge Ferrer
  */
-@Component(
-	immediate = true, property = "liferay.vulcan.context.provider=true",
-	service = CompanyContextProvider.class
-)
+@Component(immediate = true, property = "liferay.vulcan.context.provider=true")
```
Comment (JorgeFerrer) [inline vulcan-liferay-api/src/main/java/com/liferay/vulcan/liferay/internal/provider/CompanyContextProvider.java]: Could you provide some more details of this change?

Comment (ahdezma) [inline vulcan-liferay-api/src/main/java/com/liferay/vulcan/liferay/internal/provider/CompanyContextProvider.java]: Before this change `CompanyContexProvider` wasn't being injected in the application since `VulcanApplication` is injecting `ContextProvider` and we were exposing it as a `CompanyContextProvider` (which is an internal class)

File: vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/internal/InvalidGenericException.java (hunk 1 above)
Comment (ahdezma) [inline vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/internal/InvalidGenericException.java]: I don't mind it, but I fear that if we follow that approach, in the end we end up having methods like the ones in portal, which throw `PortalException` and then developers needs to inspect the implementation to see the actual exceptions it throw.

However, whether we take one road or another, for this type of cases, where we just want to crash the application with a RuntimeException, I think this approach is clearer than the nested one.

File: vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/RepresentorManager.java (hunk 2 above)
Comment (ahdezma) [inline vulcan-wiring-osgi/src/main/java/com/liferay/vulcan/wiring/osgi/RepresentorManager.java]: Same as the others, getting one element from that map returns a list, not an embeddedTuple.
